pkg/bff/gds.go

```diff
@@ -107,8 +108,15 @@ func (s *Server) Lookup(c *gin.Context) {
 		return rep, nil
 	}
 
+	ctx := c.Request.Context()
+	if s.conf.Sentry.TrackPerformance {
+		// Track Lookup request performance
+		span := sentry.StartSpan(ctx, "Lookup", sentry.TransactionName(req.String()))
+		defer span.Finish()
+	}
```

Comment:
This is to verify that the perf tracking works, but apparently we can nest spans for more fine-grained perf tracking if we need it.

Reply:
Very cool! Is it possible to embed this as a middleware to make it easy to track all API requests, or is it something we want to control at a pretty fine granularity?

Reply:
We can definitely do this, the question is what do we want to label the transactions with? It's a bit difficult to use the request body from the middleware but should we just use a hard coded label or the content length for now?

Reply:
Is the label the `"Lookup"` string in this code or the `sentry.TransactionName(req.String())` argument? 

If you're looking for the `"Lookup"` label, then the path of the request, e.g. `/v1/lookup` would be a good label -- and this should be available in the header, I haven't played around with the Gin context too much, but it's definitely in the HTTP metadata, so I assume it's easily available via Gin. 

I'm not 100% sure I understand the `req.String()` part of this, but if a unique, per-request ID is required; one thing that I used to do when I was writing raw HTTP servers was to add a UUID to the request context as it was coming into the server -- inner middleware and the handler could access that request ID if they wanted, and then you can use it on the way out of the handler. E.g. most middleware is structured as:

```go
func Middleware(req *http.Request, next http.HandlerFunc) (*http.Response, error) {
    // Handle incoming request before other middleware and handler
    // Add unique ID here
    
    rep, err = next(req) 

    // Handle outgoing request 
}
```

I know Gin complicates this a bit with its context mechanism and I don't recall the structure of Gin middleware off the top of my head, but that could be a solution. 

Also, I seem to remember that Sentry allows the front end to specify Transaction IDs so that we can track an error on both the front-end and the backend. E.g. if there is something like an `X-SENTRY-XFER-ID` header that the front-end sets we could use that as the unique request ID to match the profiling across services. 

Sorry if I didn't totally get the question - hope this is close to what you were asking at least.

Reply:
So when you log a transaction you have to give it an "operation" name which would correspond to the request path in our case, and optionally you can pass in a label that appears in the Sentry dashboard. It seems that Sentry is generating transaction ids anyway so I think that part is just a mapping from the human-interpretable label to the id. So I think I'll just leave that option blank for now until we have a better idea for it.

![Screen Shot 2022-05-05 at 4 48 15 PM](https://user-images.githubusercontent.com/42919891/167031140-ddf8282e-db4a-4f9d-909f-ce2010b80703.png)

Reply:
ok sounds good; sorry I didn't totally understand what `req.String()` was doing, but it sounds like we have a path forward and we can review when we start getting actual traces.